service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractRequestHandler.java

```diff
@@ -141,19 +141,20 @@ private String getRequiredValue(Document xml, String xpath) {
     public void handleContinue(String conversationId, String payload) {
         if (payload.contains(CONTINUE_ACKNOWLEDGEMENT)) {
             var ehrExtractStatus = ehrExtractStatusService.updateEhrExtractStatusContinue(conversationId);
-            ehrExtractStatus
-                .getGpcAccessDocument()
-                .getDocuments()
-                .forEach(gpcDocument -> createContinueTasks(ehrExtractStatus, gpcDocument.getObjectName()));
+            var documents = ehrExtractStatus.getGpcAccessDocument().getDocuments();
+            for (int documentPosition = 0; documentPosition < documents.size(); documentPosition++) {
+                createContinueTasks(ehrExtractStatus, documents.get(documentPosition).getObjectName(), documentPosition);
+            }
         } else {
             throw new InvalidInboundMessageException("Continue Message did not have Continue Acknowledgment, conversationId: "
                 + conversationId);
         }
     }
 
-    private void createContinueTasks(EhrExtractStatus ehrExtractStatus, String documentName) {
-        var sendEhrContinueTaskDefinition = SendEhrContinueTaskDefinition.builder()
+    private void createContinueTasks(EhrExtractStatus ehrExtractStatus, String documentName, int documentLocation) {
+        var sendEhrContinueTaskDefinition = SendEhrCommonTaskDefinition.builder()
```

Comment:
This naming might be more understandable:

```suggestion
    private void createSendDocumentTasks(EhrExtractStatus ehrExtractStatus, String documentName, int documentLocation) {
        var sendDocumentTaskDefinition = SendDocumentTaskDefinition.builder()
```